lib/loader.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+const fs = require('fs');
+const fetch = require('node-fetch');
+const yaml = require('js-yaml');
+const resolver = require('./resolver.js');
+
+class ExtendableError extends Error {
+  constructor(message) {
+    super(message);
+    this.name = this.constructor.name;
+  }
+}
+
+class OpenError extends ExtendableError {}
+class ReadError extends ExtendableError {}
+
+function readFileAsync(filename, encoding) {
+    return new Promise((resolve, reject) => {
+        fs.readFile(filename, encoding, (err, data) => {
+            if (err)
+                reject(err);
+            else
+                resolve(data);
+        });
+    });
+}
+
+function readSpecFile(file) {
+    if (file && file.startsWith('http')) {
+        return fetch(file).then(res => {
+            if (res.status !== 200) {
+                throw new Error(`Received status code ${res.status}`);
+            }
+            return res.text();
+        })
+    }
+    else {
+        return readFileAsync(file, 'utf8').then(data => data);
+    }
+}
+
+const resolveContent = (openapi, source) => {
+    const options = {
+        resolve: true,
+        cache: [],
+        externals: [],
+        externalRefs: {},
+        rewriteRefs: true,
+        origin: source,
+        source: source
+    };
+
+    options.openapi = openapi;
```

Comment:
Why not ?
```
const options = {
    resolve: true,
    cache: [],
    externals: [],
    externalRefs: {},
    rewriteRefs: true,
    origin: source,
    source: source,
    openapi : openapi
};
```

Reply:
Ha yes. This was code that changed over time, thanks for catching it.